sympy/solvers/diophantine/diophantine.py

```diff
@@ -2174,6 +2174,25 @@ def cornacchia(a:int, b:int, m:int) -> set[tuple[int, int]]:
     """
     # Assume gcd(a, b) = gcd(a, m) = 1 and a, b > 0 but no error checking
     sols = set()
+
+    if a + b > m:
+        # xy = 0 must hold if there exists a solution
+        if m % a == 0:
```

Comment:
Given the constraint `gcd(a, m) = 1`, I think the condition `if a == 1` should be sufficient.